yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -497,6 +497,20 @@ private[spark] class Client(
    */
   private def createConfArchive(): File = {
     val hadoopConfFiles = new HashMap[String, File]()
+
+    // Uploading $SPARK_CONF_DIR/log4j.properties file to the distributed cache to make sure that
+    // the executors will use the latest configurations instead of the default values. This is
+    // required when user changes log4j.properties directly to set the log configurations. If
+    // configuration file is provided through --files then executors will be taking configurations
+    // from --files instead of $SPARK_CONF_DIR/log4j.properties.
+    Option(Utils.getContextOrSparkClassLoader.getResource("log4j.properties"))
+      .map(_.getPath).map(path => {
```

Comment:
Technically, you should only look at this URL if its protocol is `file`, otherwise even if the path matches a local file, it wouldn't be correct.

Also, style:

    .map { path =>
      // code
    }